### ISSUE TITLE
JSON corrupting with change of object value

### ISSUE THREAD

**Issue by turagittech** [SQLPlayer/azure.datafactory.tools#33]
**Describe the bug**
Deploying to a production environment and using the ability to change object values to update a web URL in the pipeline
Once the code is deployed mappings in the pipeline are missing. We have tried with and without to evaluate the issue and get it to work. Deploys ok without URL change mappings are present. Use config file to create URL change and the problem occurs

**To Reproduce**
Your script to reproduce the behaviour:
```
Added this to the config-Production.csv (URL has been modified to protect a system)
pipeline,GetFusebillCustomers_ToBlob,activities[0].typeProperties.url,https://abcheg-prd-it-akv-001.vault.azure.net/secrets/secretKey/9e0e3e8561b040f7c34d4ffa0u983a78?api-version=7.0
Ran Publish-AdfV2FromJson  `
>>     -RootFolder C:\Users\peter\source\repos\MBQLD\Data-Pipelines `
>>     -ResourceGroupName "turagit-data-01" `
>>     -DataFactoryName "TGT-PRD-001" `
>>     -Location "australiaeast" `
>>     -Stage "Production"
(This is not a production system in this test)

Your script here
Ran the command from Powershell on my PC to obtain the changed file to be able to see the source of the issue
```

**Expected behaviour**
URL to update and all of the JSON would deploy objects successfully
I can provide a copy of the full JSON, but concerned they might leak information

**Screenshots**
![MicrosoftTeams-image (4)](https://user-images.githubusercontent.com/4639804/92377626-c2b78300-f147-11ea-83f4-412c5967e68c.png)
![MicrosoftTeams-image (5)](https://user-images.githubusercontent.com/4639804/92377630-c3501980-f147-11ea-9f4d-b520cd6eed14.png)

**Version of PowerShell module of:**
 - azure.datafactory.tools [e.g.0.10.1]
Version 0.14.000

**Comment by NowinskiK** [comment]
Thanks, Peter for raising this issue. I will take a look at this as soon as I can.

**Comment by NowinskiK** [comment]
Hi Peter. Thanks again for reporting this very precisely.  
Could you attach the JSON file of the pipeline?  
Please, remove all other activities and leave only the first one (which contains a mapping) if you have any information you would prefer not to share here.

**Comment by pjtaylor** [comment]
FYI, I have the same issue should another guinea pig be required. Will leave to Peter for now.

**Comment by pjtaylor** [comment]
Hi Kamil, here is my example.
[Issue33.zip](https://github.com/SQLPlayer/azure.datafactory.tools/files/5202456/Issue33.zip)

**Comment by turagittech** [comment]
Hi Kamil
Apologies for the late response, been closing out a project

Here is a file. The key vault URL has been altered to not identify the one in use
[Example Issue33.zip](https://github.com/SQLPlayer/azure.datafactory.tools/files/5251613/Example.Issue33.zip)

**Comment by NowinskiK** [comment]
I was looking for this and I know where it comes from, but unfortunately, it's not an easy-to-fix issue.

**Comment by pjtaylor** [comment]
No problem. Thanks for the update, Kamil.

**Comment by turagittech** [comment]
Thanks for letting us know, hopefully, a solution will arise

**Comment by NowinskiK** [comment]
Apparently, this issue has been fixed by merge #55. It will be released in nearest version 0.18.
